cmd/virtual-kubelet/commands/root/root.go

```diff
@@ -60,6 +61,9 @@ This allows users to schedule kubernetes workloads on nodes that aren't running
 }
 
 func runRootCommand(ctx context.Context, c Opts) error {
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
This seems like it'd be very confusing to troubleshoot a deadlock. Why not explicitly move it?

Reply:
For sake of simplicity: wait in one place instead of each return.
Another option would be to use wait group for goroutine with `node.Run` only, but leaving the rest goroutines without attention is not nice, I think.